sphinx/util/docfields.py

```diff
@@ -254,8 +254,12 @@ def handle_item(fieldarg: str, content: str) -> nodes.paragraph:
                 else:
                     par += fieldtype
                 par += nodes.Text(')')
-            par += nodes.Text(' -- ')
-            par += content
+            empty = len(content) == 0
```

Comment:
If "content" is not a string nor a list of nodes but a real `Node`, you can write:

```
if content.astext().strip():
    par += nodes.Text(' -- ')
    par += content
```

If content is a list of string, I don't think it's right to just check the first node element being added since the next nodes in the list might be non-empty text.